cmd/osmosisd/cmd/root.go

```diff
@@ -623,17 +635,29 @@ adaptive-fee-enabled = "true"
 [osmosis-sqs]
 
 # SQS service is disabled by default.
-is-enabled = "false"
+is-enabled = "{{ .SidecarQueryServerConfig.IsEnabled }}"
 
 # The hostname of the GRPC sqs service
 grpc-ingest-address = "{{ .SidecarQueryServerConfig.GRPCIngestAddress }}"
 # The maximum size of the GRPC message that can be received by the sqs service in bytes.
 grpc-ingest-max-call-size-bytes = "{{ .SidecarQueryServerConfig.GRPCIngestMaxCallSizeBytes }}"
 
 ###############################################################################
-###              		       Wasm Configuration    					    ###
+###                            Wasm Configuration                           ###
 ###############################################################################
-` + wasmtypes.DefaultConfigTemplate()
+
+[wasm]
+# Smart query gas limit is the max gas to be used in a smart query contract call
+query_gas_limit = {{ .WasmConfig.SmartQueryGasLimit }}
+
+# in-memory cache for Wasm contracts. Set to 0 to disable.
+# The value is in MiB not bytes
+memory_cache_size = {{ .WasmConfig.MemoryCacheSize }}
+
+# Simulation gas limit is the max gas to be used in a tx simulation call.
+# When not set the consensus max block gas is used instead
+# simulation_gas_limit =
+`
```

Comment:
We need to manually define the default config template for wasm because they define specific values instead of pointing to the config values (i.e. 3000000 instead of .WasmConfig.SmartQueryGasLimit). This means that when we use the template to overwrite the new app.toml, no matter what the viper flags are set to (from the user's preference), the template will use the hardcoded values.